ProfilerGUI/Source/Configurator/MainViewModel.cs

```diff
@@ -0,0 +1,190 @@
+
+        /// <summary>
+        /// Summary of the configured upload.
+        /// </summary>
+        public string UploadSummary
+        {
+            get
+            {
+                if (uploadConfig == null)
+                {
+                    return "No upload";
+                }
+                if (uploadConfig.Teamscale != null)
+                {
+                    return $"Upload to {uploadConfig.Teamscale}";
+                }
+                return $"Upload to directory {uploadConfig.Directory}";
+            }
+        }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="launchTargetAppDirectly">If this is set, then the target application is started right away.
+        /// A default config must exist for this to do anything.</param>
+        public MainViewModel(bool launchTargetAppDirectly)
+        {
+            ProfilerConfiguration configuration = new ProfilerConfiguration();
+            if (File.Exists(ProfilerConfiguration.ConfigFilePath))
+            {
+                logger.Info("Loaded config from {configPath}", ProfilerConfiguration.ConfigFilePath);
+                configuration = ProfilerConfiguration.ReadFromFile();
+
+                if (launchTargetAppDirectly)
+                {
+                    RunProfiledApplication();
+                }
+            }
+
+            TargetApp = new TargetAppModel(configuration);
+            TargetApp.PropertyChanged += (sender, args) =>
+            {
+                UiUtils.ReRaise(PropertyChanged, this, nameof(TargetApp), args);
+
+                if (args.PropertyName == nameof(TargetApp.ApplicationType))
+                {
+                    UiUtils.Raise(PropertyChanged, this, nameof(SelectedBitnessIndex));
+                }
+            };
+
+            uploadConfig = ReadUploadConfigFromDisk();
+        }
+
+        /// <summary>
+        /// Shows the dialog to configure the trace upload.
+        /// </summary>
+        internal void OpenUploadConfigDialog()
+        {
+            UploadConfigWindow dialog = new UploadConfigWindow(uploadConfig);
+            bool? result = dialog.ShowDialog();
+            if (result == true)
+            {
+                uploadConfig = dialog.Config;
+            }
+            UiUtils.Raise(PropertyChanged, this, nameof(UploadSummary));
+        }
+
+        /// <summary>
+        /// Resets the fields that describe the application that should be profiled.
+        /// </summary>
+        internal void ClearTargetAppFields()
+        {
+            TargetApp.Clear();
+        }
+
+        /// <summary>
+        /// Saves the current configuration to a file.
+        /// </summary>
+        internal void SaveConfiguration()
+        {
+            try
+            {
+                TargetApp.Configuration.WriteToFile();
+                logger.Info("Wrote config to {configPath}", ProfilerConfiguration.ConfigFilePath);
+                SaveUploadConfig();
```

Comment:
At first I though the error message below is incomplete, because it doesn't cover the "failed to save upload config" case. Then I realised that `SaveUploadConfig()` does it's own error handling. Move the call out of the try catch?

Also: Inconsistent naming `Config` vs. `Configuration`